### PR TITLE
X509Certificate: Get rid of deprecated OpenSSL 1.1 APIs

### PR DESCRIPTION
**Pull request by neheb**
Added a missing implicit header and added compatibility macros.

Signed-off-by: Rosen Penev <rosenp@gmail.com>

### CODE REVIEW

**Comment by obiltschnig** [comment]
Please re-target your PR against poco-1.10.0 branch.

**Comment by neheb** [comment]
I think that should do it.

File: Crypto/src/X509Certificate.cpp [hunk 1]
```diff
@@ -255,7 +261,7 @@ void X509Certificate::extractNames(std::string& cmnName, std::set<std::string>&
 			const GENERAL_NAME* name = sk_GENERAL_NAME_value(names, i);
 			if (name->type == GEN_DNS)
 			{
-				const char* data = reinterpret_cast<char*>(ASN1_STRING_data(name->d.ia5));
+				const char* data = reinterpret_cast<char*>(ASN1_STRING_get0_data(name->d.ia5));
```
Comment (neheb) [inline Crypto/src/X509Certificate.cpp]: My C++ is poor but note that ASN1_STRING_get0_data returns const unsigned char* whereas ASN1_STRING_data returns unsigned char*. There might be a const_cast needed.

Comment (obiltschnig) [reply Crypto/src/X509Certificate.cpp]: Thanks, will fix.

**Pull request merged**